lib/apiservers/engine/backends/cache/image_cache.go

```diff
@@ -66,8 +66,6 @@ func NewImageCache() *ImageCache {
 
 // Update adds new images to the cache
 func (c *ImageCache) Update(client *client.PortLayer) error {
-	c.m.Lock()
-	defer c.m.Unlock()
 
 	CacheNotUpdated = true
```

Comment:
By the way, removing this lock caused this var to become a data race :) You can something else here to protect that (channel, atomic var etc (for atomic see https://github.com/vmware/vic/blob/master/lib/dns/cache.go))